impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Gauge.js

```diff
@@ -23,46 +23,130 @@ define([
 
   return BaseView.extend(module.id, {
 
+    _buildAxisLineColors: function () {
+
+      var colorArray = this._getContinuousColorScale();
+
+      var step = 1 / colorArray.length;
+      var threshold = step;
+
+      var axisLineColors = [];
+
+      for (var i = 0; i < colorArray.length; i++) {
+        axisLineColors.push([threshold, colorArray[i]]);
+        threshold += step;
+      }
+
+      return axisLineColors;
+    },
+
+
     _buildSeries: function(echartData) {
-      var dataTable = this.model.data;
+      var model = this.model;
+      var dataTable = model.data;
```

Comment:
The following is just a slight improvement, moving some vars inward.
Doing it more concisely would require going "generic" in the directions and dimensions variables, and I admit it would be harder...

```js
_buildData: function() {
      var model = this.model;
      var dataTable = model.data;

      if(dataTable.originalCrossTable) {
        dataTable = dataTable.originalCrossTable.toPlainTable({skipRowsWithAllNullMeasures: true});
      }

      var rowCount = dataTable.getNumberOfRows();
      var categoriesColIndexes = model.rows.fieldIndexes;
      var measureColIndex = model.measures.fieldIndexes[0];
      var font = util.getDefaultFont(null, 12);
      var legendPosition = model.legendPosition || "right";
      var bgColorDetail = this._buildColors();

      var records = [];

      var bandPercent = 0.95;

      var self = this;

      function pushRecordsHorizontal(offsetCenterYPercent){
        var offsetCenterX;
        var innerWidth = model.width * 0.8;
        var stepWidthHorizontal = innerWidth / (rowCount - (1 - bandPercent));
        var bandWidthHorizontal = bandPercent * stepWidthHorizontal;
        var detailWidthHorizontal = stepWidthHorizontal * 0.7;
        var offsetCenterX0 = -innerWidth / 2 + bandWidthHorizontal / 2;

        for (var i = 0; i < rowCount; i++) {
          offsetCenterX = offsetCenterX0 + stepWidthHorizontal * i

          records.push({
            name: dataTable.getCompositeFormattedValue(i, categoriesColIndexes, self.groupedLabelSeparator),
            value: dataTable.getValue(i, measureColIndex),
            tooltip: self._buildTooltip(self._buildRowTooltipHtml(dataTable, i), font),
            title: {
              offsetCenter: [offsetCenterX, offsetCenterYPercent + "%"],
              overflow: "truncate",
              width: bandWidthHorizontal
            },
            detail: {
              offsetCenter: [offsetCenterX, (offsetCenterYPercent + 15) + "%"],
              width: detailWidthHorizontal,
              backgroundColor: bgColorDetail[i]
            },
            visualKey: dataTable.getCompositeValue(i, categoriesColIndexes, self.groupedLabelSeparator)
          });
        }
      }

      function pushRecordsVertical(offsetCenterXPercent){
        var offsetCenterY;
        var innerHeight = model.height * 0.8;
        var stepWidthVertical = innerHeight / (rowCount - (1 - bandPercent));
        var bandWidthVertical = bandPercent * stepWidthVertical;
        var detailWidthVertical = stepWidthVertical * 0.8;
        var offsetCenterY0 = -innerHeight / 2 + bandWidthVertical / 2;

        for (var i = 0; i < rowCount; i++) {
          offsetCenterY = offsetCenterY0 + stepWidthVertical * i;

          records.push({
            name: dataTable.getCompositeFormattedValue(i, categoriesColIndexes, self.groupedLabelSeparator),
            value: dataTable.getValue(i, measureColIndex),
            tooltip: self._buildTooltip(self._buildRowTooltipHtml(dataTable, i), font),
            title: {
              offsetCenter: [offsetCenterXPercent + "%", offsetCenterY],
              overflow: "truncate",
              width: bandWidthVertical
            },
            detail: {
              offsetCenter: [(offsetCenterXPercent + 40) + "%", offsetCenterY],
              width: detailWidthVertical,
              backgroundColor: bgColorDetail[i]
            },
            visualKey: dataTable.getCompositeValue(i, categoriesColIndexes, self.groupedLabelSeparator)
          });
        }
      }

      switch (legendPosition) {
        case "bottom":
          pushRecordsHorizontal(100);
          break;

        case "top":
          pushRecordsHorizontal(-135);
          break;

        case "left":
          pushRecordsVertical(-235);
          break;

        case "right":
        default:
          pushRecordsVertical(150);
      }

      return records;
    }
```